File: 15.py

def lattice(n,m):
    # if n == 1 :
    #     return 1
    # elif n == 13 and m == 13:
    #     return 2704156
    # else:
    #     total = 0
    #     for i in range(1, m+1):
    #         total += lattice(n-1, m-i+1)
    #     return total
    lat = [[1 for i in range(m)] for i in range(n)]
    for i in range(n-2,-1,-1):
        for j in range(m-2,-1, -1):
            lat[i][j] = lat[i+1][j] +lat[i][j+1]



    return lat[0][0]

print(lattice(21,21))

File: cpp/15.cpp
#include <iostream>
#include <vector>

using namespace std;

long int lattice(int n, int m){
    vector<vector<long int>> lat {n, vector<long int> (m, 1)};
    for (int i = n-2; i >= 0; i--)
    {
        for (int j = m-2; j >= 0; j--)
        {
            lat[i][j] = lat[i+1][j] + lat[i][j+1];
        }
    }
    return lat[0][0];
}

int main() {
    
    cout << lattice(21, 21) << endl;

    return 0;
}